scripts/build-plugins-report.js

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+const open = require('open');
+const { execFile: execFileCb } = require('child_process');
+const { promisify } = require('util');
+
+const execFile = promisify(execFileCb);
+
+const rootDirectory = path.resolve(__dirname, '..');
+
+async function run(command, ...args) {
+  const { stdout, stderr } = await execFile(command, args, {
+    cwd: rootDirectory,
+  });
+
+  if (stderr) {
+    console.error(stderr);
+  }
+
+  return stdout.trim();
+}
+
+function generateHtmlReport(rows) {
+  const thead = `<tr>${rows[0].map(cell => `<th>${cell}</th>`).join('')}</tr>`;
+
+  const tbody = rows
+    .slice(1)
+    .map(row => `<tr>${row.map(cell => `<td>${cell}</td>`).join('')}</tr>`)
+    .join('');
+
+  return `
+    <html lang="en">
+      <head>
+        <title>Backstage Plugins Report</title>
+        <style>
+          * {font-family:sans-serif;}
+          table {width:100%; border-collapse: collapse;}
+          table,td,th {border:1px solid;}
+        </style>
+      </head>
+      <body>
+        <table>
+          <thead>${thead}</thead>
+          <tbody>${tbody}</tbody>
+        </table>
+      </body>
+    </html>
+  `;
+}
+
+const PLUGIN_ROLES = ['frontend-plugin', 'backend-plugin'];
+
+async function main() {
+  const pluginsDirectory = path.resolve(rootDirectory, 'plugins');
+
+  const directoryNames = fs
+    .readdirSync(pluginsDirectory, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name);
+
+  const tableRows = [['Plugin', 'Author', 'Message', 'Date']];
+
+  for (const directoryName of directoryNames) {
+    console.log(`🔎 Reading data for ${directoryName}`);
+
+    const directoryPath = path.resolve(pluginsDirectory, directoryName);
+
+    const packageJson = await fs.readJson(
+      path.resolve(directoryPath, 'package.json'),
+    );
+
+    const plugin = PLUGIN_ROLES.includes(packageJson?.backstage?.role ?? '');
+
+    if (!plugin) continue;
+
+    let data;
+    let skip = 0;
+    const maxCount = 10;
+    while (!data && skip <= 100) {
+      const output = await run(
+        'git',
+        'log',
+        'origin/master',
+        `--skip=${skip}`,
+        `--max-count=${maxCount}`,
+        '--format=%an;%s;%as',
+        directoryPath,
+      );
+
+      const lines = output.split('\n');
+
+      data = lines.find(line => {
+        // exclude merge commits
+        if (line.includes('Merge pull request #')) {
+          return false;
+        }
+
+        // exclude commits authored by a bot
+        if (
+          line.startsWith('renovate[bot]') ||
+          line.startsWith('github-actions[bot]')
+        ) {
+          return false;
+        }
+
+        return true;
```

Comment:
Maybe exclude maintainers' commits as well? I could also expose a regex flag if that works best 🙂

Reply:
I did the first and third refinements, and the data seems to be more accurate, I'll try out the second one.

Reply:
Done 🙌🏻 : https://github.com/backstage/backstage/pull/20487/commits/352ef85d7f74fdb5ae0292972cd91fc0424f6c10